ts/packages/anchor/src/coder/borsh/event.ts

```diff
@@ -70,4 +99,59 @@ export class BorshEventCoder implements EventCoder {
     const data = layout.decode(logArr.slice(8));
     return { data, name: eventName };
   }
+
+  /**
+   * Check the log data to be transaction CPI event:
+   * Expected data format:
+   *  < cpi event discriminator | event name discriminator | event data >
+   * If matches cpi event discriminator
+   * < event name | event data> base64 formatted is returned
+   * otherwise null is returned.
+   */
+  parseAsTransactionCpiData(log: string): string | null {
+    let encodedLog: Buffer;
+    try {
+      // verification if log is transaction cpi data encoded with base58
+      encodedLog = bs58.decode(log);
+    } catch (e) {
+      return null;
+    }
+    const disc = encodedLog.slice(0, 8);
+    if (disc.equals(BorshEventCoder.eventIxTag.toBuffer("le"))) {
+      // after CPI tag data follows in format of standard event
+      return base64.encode(encodedLog.slice(8));
+    } else {
+      return null;
+    }
+  }
+
+  public parseCpiEvents(
+    transactionResponse: VersionedTransactionResponse
```

Comment:
It would be great if we could also support `TransactionResponse`.

Reply:
Yes, I added the `TransactionResponse` to types.

When autogenerated, how can I run it that I can remove the extended class and make it as generated? I run the `./setup-tests.sh` but I haven't found a guideline on development that can help me.